sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -0,0 +1,391 @@
+        glm.fit(X, y)
+
+
+@pytest.mark.parametrize('copy_X', ['not bool', 1, 0, [True]])
+def test_glm_copy_X_argument(copy_X):
+    """Test GLM for invalid copy_X arguments."""
+    y = np.array([1, 2])
+    X = np.array([[1], [1]])
+    glm = GeneralizedLinearRegressor(copy_X=copy_X)
+    with pytest.raises(ValueError, match="copy_X must be bool"):
+        glm.fit(X, y)
+
+
+@pytest.mark.parametrize('check_input', ['not bool', 1, 0, [True]])
+def test_glm_check_input_argument(check_input):
+    """Test GLM for invalid check_input argument."""
+    y = np.array([1, 2])
+    X = np.array([[1], [1]])
+    glm = GeneralizedLinearRegressor(check_input=check_input)
+    with pytest.raises(ValueError, match="check_input must be bool"):
+        glm.fit(X, y)
+
+
+@pytest.mark.parametrize('solver', GLM_SOLVERS)
+def test_glm_identity_regression(solver):
+    """Test GLM regression with identity link on a simple dataset."""
+    coef = [1., 2.]
+    X = np.array([[1, 1, 1, 1, 1], [0, 1, 2, 3, 4]]).T
+    y = np.dot(X, coef)
+    glm = GeneralizedLinearRegressor(alpha=0, family='normal', link='identity',
+                                     fit_intercept=False, solver=solver,
+                                     tol=1e-7)
+    res = glm.fit(X, y)
+    assert_allclose(res.coef_, coef, rtol=1e-6)
+
+
+@pytest.mark.parametrize(
+    'family',
+    [NormalDistribution(), PoissonDistribution(),
+     GammaDistribution(), InverseGaussianDistribution(),
+     TweedieDistribution(power=1.5), TweedieDistribution(power=4.5)])
+@pytest.mark.parametrize('solver, tol', [('lbfgs', 1e-6)])
+def test_glm_log_regression(family, solver, tol):
+    """Test GLM regression with log link on a simple dataset."""
+    coef = [0.2, -0.1]
+    X = np.array([[1, 1, 1, 1, 1], [0, 1, 2, 3, 4]]).T
+    y = np.exp(np.dot(X, coef))
+    glm = GeneralizedLinearRegressor(
+                alpha=0, family=family, link='log', fit_intercept=False,
+                solver=solver, tol=tol)
+    res = glm.fit(X, y)
+    assert_allclose(res.coef_, coef, rtol=5e-6)
+
+
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_warm_start(fit_intercept):
+    n_samples, n_features = 100, 10
+    n_predict = 10
+    X, y, coef = make_regression(n_samples=n_samples+n_predict,
+                                 n_features=n_features,
+                                 n_informative=n_features-2, noise=0.5,
+                                 coef=True, random_state=42)
+
+    glm1 = GeneralizedLinearRegressor(
+        warm_start=False,
+        fit_intercept=fit_intercept,
+        max_iter=1000
+    )
+    glm1.fit(X, y)
+
+    glm2 = GeneralizedLinearRegressor(
+        warm_start=True,
+        fit_intercept=fit_intercept,
+        max_iter=1
+    )
+    glm2.fit(X, y)
+    assert glm1.score(X, y) > glm2.score(X, y)
+    glm2.set_params(max_iter=1000)
+    glm2.fit(X, y)
+    assert_allclose(glm1.coef_, glm2.coef_, rtol=1e-4, atol=1e-5)
```

Comment:
I'm surprised that they don't exactly match? Is the solver stochastic?

Reply:
> I'm surprised that they don't exactly match? Is the solver stochastic?

It's not stochastic but are you sure that running lbfgs for 2*N iterations is strictly equivalent to running it twice for `N` iterations and re-using previous parameters? I don't think so, if only because the history of evaluated points for the approximate hessian estimation is not going to be the same when you restart. The actual tolerance needed here is `rtol=1e-5` which seems good enough to me.

Reply:
that makes sense (I know nothing about lbfgs ;) )
maybe a small comment would be useful